pipelines/tekton/aiedge-e2e/templates/credentials-s3.secret.yaml.template

```diff
@@ -7,5 +7,8 @@ metadata:
     app.kubernetes.io/part-of: rhoai-edge-pipelines
 
 stringData:
+  # The endpoint_url property is an optional one when using AWS S3, generally it takes priority over the region when specified. 
+  # If the endpoint_url is not specified it would default to an AWS endpoint based on region specified.
+  # Set the bucket region correctly.
```

Comment:
So the region is actually based on the credentials? And in case a user wants to use a bucket from a different region than credentials they would specify an endpoint_url from a different region?
In other words, `endpoint_url` is optional while `region` is mandatory?

Reply:
Thanks. How does it relate to the JIRA then? There it says `Currently the PipelineRuns will fail in the fetch-model-s3 step if the S3 bucket region is not the same region as the IAM credentials.` Based on what you said the S3 will intelligently handle this, if I specify a region us-west but bucket is in the us-east, it will be handled with an automatic redirect. So the S3 region doesn't have to be correct one but it has to be one from the regions provided by AWS. If that is so, I would reword the comment then.

Reply:
@MarianMacik I think the Jira issue wording was based on lack of complete understanding of the problem, and was likely based on always having the endpoint_url set. So if you had config where `region` was set to `eu-west-1` and corresponding endpoint_url of `https://s3.eu-west-1.amazonaws.com`, but the bucket was in `us-east-2`, then you'd get the behaviour described in the issue in Jira.

Reply:
>I think the Jira issue wording was based on lack of complete understanding of the problem, and was likely based on always having the endpoint_url set. 

Correct.  The original Jira was created based on my experience with boto3 connections to non-AWS S3 compatible endpoints where the endpoint_url AND [default] region where required for all client sessions.

Acceptance Criteria for this PR is that it should be clear to the user that `endpoint_url` and `region` are required for all non-AWS object storage servers BUT AWS S3 only requires the region